addons/point_of_sale/models/pos_session.py

```diff
@@ -300,15 +305,485 @@ def _validate_session(self):
+        account_id, sign, tax_keys = key
+        tax_ids = [tax[0] for tax in tax_keys]
+        applied_taxes = self.env['account.tax'].browse(tax_ids)
+        name = '' if not applied_taxes else '%s group: %s' % ('Sales' if sign == 1 else 'Refund', ', '.join([tax.name for tax in applied_taxes]))
+        base_tags = applied_taxes\
+            .mapped('invoice_repartition_line_ids' if sign == 1 else 'refund_repartition_line_ids')\
+            .filtered(lambda line: line.repartition_type == 'base')\
+            .tag_ids
+        partial_vals = {
+            'name': name,
+            'account_id': account_id,
+            'move_id': self.move_id.id,
+            'tax_ids': [(6, 0, tax_ids)],
+            'tag_ids': [(6, 0, base_tags.ids)],
+        }
+        return self._credit_amounts(partial_vals, amount, amount_converted)
+
+    def _get_tax_vals(self, key, amount, amount_converted, base_amount):
+        account_id, repartition_line_id, tax_id, tag_ids = key
+        tax = self.env['account.tax'].browse(tax_id)
+        partial_args = {
+            'name': tax.name,
+            'account_id': account_id,
+            'move_id': self.move_id.id,
+            'tax_base_amount': base_amount,
+            'tax_repartition_line_id': repartition_line_id,
+            'tag_ids': [(6, 0, tag_ids)],
+        }
+        return self._credit_amounts(partial_args, amount, amount_converted)
+
+    def _get_stock_expense_vals(self, exp_account, amount, amount_converted):
+        partial_args = {'account_id': exp_account.id, 'move_id': self.move_id.id}
+        return self._debit_amounts(partial_args, amount, amount_converted)
+
+    def _get_stock_output_vals(self, out_account, amount, amount_converted):
+        partial_args = {'account_id': out_account.id, 'move_id': self.move_id.id}
+        return self._credit_amounts(partial_args, amount, amount_converted)
+
+    def _get_statement_line_vals(self, statement, receivable_account, amount, amount_converted):
+        return {
+            'date': fields.Date.context_today(self),
+            'amount': amount,
+            'name': self.name,
+            'statement_id': statement.id,
+            'account_id': receivable_account.id,
+        }
+
+    def _update_amounts(self, new_amount, date, amount, amount_converted):
+        return {
+            'amount': amount + new_amount,
+            'amount_converted': amount_converted if self.is_in_company_currency else (amount_converted + self._amount_converter(new_amount, date)),
+        }
+
+    def _update_tax_amounts(self, new_amount, new_base_amount, date, amount, amount_converted, base_amount):
+        return {
+            'amount': amount + new_amount,
+            'amount_converted': amount_converted if self.is_in_company_currency else (amount_converted + self._amount_converter(new_amount, date)),
+            'base_amount': base_amount + new_base_amount
+        }
+
+    def _get_rounding_error_vals(self):
```

Comment:
I don't think this method should exist. There should not have differences between the sales and receivable. The rounding is supposed to be the same in frontend than in backend.

Reply:
I am sure that I was able to reproduce this rounding issue multiple times. Let me try to reproduce it again and see how it goes.